packages/hir/src/lowering/types.candy

```diff
@@ -46,23 +46,36 @@ fun astInlineTypeToHirInlineType(
             .toSet() as Set<HirType | HirTrait | HirFunction | HirModule>)
         }
 
-        // There should only be one candidate left.
+        // There should be exactly one candidate left.
+        if (candidates as Iterable).isEmpty() {
+          return Tuple(
+            HirErrorType(),
+            List.of1<CompilerError>(CouldNotResolveTypeCompilerError(astType as AstNamedType)),
+          )
+        }
+        if (candidates as Iterable).length() > 1 {
+          return Tuple(
+            HirErrorType(),
+            List.of1<CompilerError>(
+              //       // This cast from `Set<HirType | HirTrait | HirFunction | HirModule>` to
+              //       // `Set<HirType | HirTrait | HirModule>`succeeds because the name is uppercase, so no
+              //       // functions get resolved.
```

Comment:
```suggestion
              // This cast from `Set<HirType | HirTrait | HirFunction | HirModule>` to
              // `Set<HirType | HirTrait | HirModule>` succeeds because the name is uppercase, so no
              // functions get resolved.
```